domain/src/main/java/org/oppia/domain/profile/ProfileManagementController.kt

```diff
@@ -255,6 +282,70 @@ class ProfileManagementController @Inject constructor(
       })
   }
 
+  /**
+   * Updates the download/update on wifi only permission.
+   *
+   * @param profileId the ID corresponding to the profile being updated.
+   * @param downloadAndUpdateOnWifiOnly download and update permission on wifi only.
+   * @return a [LiveData] that indicates the success/failure of this update operation.
+   */
+  fun updateWifiPermissionDeviceSettings(
+    profileId: ProfileId,
+    downloadAndUpdateOnWifiOnly: Boolean
+  ): LiveData<AsyncResult<Any?>> {
+    val deferred = profileDataStore.storeDataWithCustomChannelAsync(updateInMemoryCache = true) {
+      val profile = it.profilesMap[profileId.internalId] ?: return@storeDataWithCustomChannelAsync Pair(
+        it,
+        ProfileActionStatus.PROFILE_NOT_FOUND
+      )
+      val profileDatabaseBuilder = it.toBuilder()
+      if (profile.isAdmin) {
+        val deviceSettingsBuilder = it.deviceSettings.toBuilder()
+        deviceSettingsBuilder.allowDownloadAndUpdateOnlyOnWifi = downloadAndUpdateOnWifiOnly
+        profileDatabaseBuilder.deviceSettings = deviceSettingsBuilder.build()
+        Pair(profileDatabaseBuilder.build(), ProfileActionStatus.SUCCESS)
+      } else {
+        Pair(profileDatabaseBuilder.build(), ProfileActionStatus.PROFILE_NOT_ADMIN)
+      }
+    }
+    return dataProviders.convertToLiveData(
+      dataProviders.createInMemoryDataProviderAsync(UPDATE_DEVICE_SETTINGS_TRANSFORMED_PROVIDER_ID) {
+        return@createInMemoryDataProviderAsync getDeferredResult(profileId, null, deferred)
+      })
+  }
+
+  /**
+   * Updates the download/update on wifi only permission.
```

Comment:
Change this comment to `Updates whether to allow automatic update of topics for admin settings.`

Reply:
Actually it is not related to Admin Settings so updated to 
`* Updates the automatically update topics permission.`

Reply:
okay got it.